### PR TITLE
test: fix unstable evict test

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve? 
 Fix unstable evict leader test. Part of https://github.com/pingcap/dm/issues/1790.

### What is changed and how it works?

When evicting a leader, we just signal the `resignCh` channel. This operation is asynchronous, so new leader may not be elected immediately after the request returns. We need to wait for a proper period of time.

https://github.com/pingcap/dm/blob/ccddd7dd71019a07695ad0d208da29a1442816e1/pkg/election/election.go#L401-L405

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: tests/ha_master/run.sh [hunk 1]
```diff
@@ -58,12 +58,17 @@ function test_evict_leader() {
...
+		  [ "$NEW_LEADER_NAME" = "$LEADER_NAME" ] && continue
+		  sleep 1
+		done
 		if [ "$NEW_LEADER_NAME" = "$LEADER_NAME" ]; then
-			echo "leader evict failed"
-			exit 1
+		  echo "leader evict failed"
+		  exit 1
 		fi
+    echo "new leader is $NEW_LEADER_NAME"
```
Comment (lance6716) [inline tests/ha_master/run.sh:71]: make fmt

**Comment by lance6716** [comment]
in my mind the behaviour is not "wait for a proper period of time", but reading the debug log of etcd I found the election key is not deleted for a long time. Is this the same thing?

File: pkg/election/election.go [hunk 2]
```diff
@@ -246,14 +246,15 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
 		elec := concurrency.NewElection(session, e.key)
 		ctx2, cancel2 := context.WithCancel(ctx)
 
+		campaignWg.Add(1)
```
Comment (lance6716) [inline pkg/election/election.go:249]: does this change has some effects?

**Comment by gozssky** [comment]
I'm not sure whether this PR can fix https://github.com/pingcap/dm/issues/1212 😂.  I add some sleep after `case <-e.resignCh`, the integration test will fail.

File: pkg/election/election.go (hunk 2 above)
Comment (gozssky) [reply pkg/election/election.go:249]: Probably not. This is a problem that I found by reading code, but the probability of this problem triggering is very small.

**Review by lance6716 — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]
LGTM

File: pkg/election/election.go (hunk 2 above)
Comment (gozssky) [reply pkg/election/election.go:249]: @lance6716  If we evict leader before line 246, can campaign be canceled successfully?

Comment (lance6716) [reply pkg/election/election.go:249]: then in line 260, the atomic variable loaded `false` and the campaign goroutine will not work

**Pull request merged**